packages/studio-web/src/app/upload/upload.component.html

```diff
@@ -128,8 +136,8 @@ <h1 i18n="Title for language box" class="title">Select your Language</h1>
                                 <mat-label i18n="Label for language dropdown">Select Language</mat-label>
                                 <mat-select [formControl]="langControl" required>
                                     <mat-option i18n="Name of undetermined language" value="und">Undetermined - (und)</mat-option>
-                                    <mat-option *ngFor="let lang of langs$ | async" [value]="lang['id']">
-                                        {{lang['name']}} - ({{lang['id']}})
+                                    <mat-option *ngFor="let lang of langs" [value]="lang.code">
+                                        {{lang.names['_']}} - ({{lang.code}})
```

Comment:
The only problem with this is that `| async` handles proper unsubscribing. I should really refactor all of our subscriptions to unsubscribe when the component is cleared though to repair memory leaks